docs(weave): Document dataset editing and deletion + cosmetics

J2-D2-3PO opened the pull request:
## Description

- Fixes https://wandb.atlassian.net/browse/DOCS-1178

In https://weave-docs.wandb.ai/guides/core-types/datasets/
- Documents dataset creation and editing flow in UI
- Touch ups to existing content

## Testing

- yarn start on local

bcsherma commented on docs/docs/guides/core-types/datasets.md:
```diff
@@ -3,15 +3,24 @@ import TabItem from '@theme/TabItem';
 # Datasets
 
-`Dataset`s enable you to collect examples for evaluation and automatically track versions for accurate comparisons. Use this to download the latest version locally with a simple API.
+Weave _datasets_ help you to organize, collect, track, and version examples for LLM application evaluation for easy comparison. You can create and interact with datasets programmatically and via the UI. 
```
I have a suspicion that `Dataset` is a proper noun in our product and should be capitalized.

andrewtruong replied:
Agree -- `Dataset` is specifically the name of an object and should be marked so!

bcsherma commented on docs/docs/guides/core-types/datasets.md:
```diff
@@ -69,55 +78,114 @@ This guide will show you how to:
+
+    ![Dataset UI](./imgs/datasetui-popout-edit-green.png)
+
+6. To add data to the new row, click the column in the new row that you want to edit. You can't edit the default **id** column in a dataset row, as this is automatically assigned by Weave upon creation. A pop-out modal for editing displays, with **Text**, **Code** and **Diff** options for formatting.
```
I find the last sentence on this line difficult to parse. Maybe `When an editable cell is clicked, a cell editor pop-out appears. The editor supports ... format options.`

bcsherma reviewed (commented):
Couple thoughts but otherwise looks good! The design team is making some styling changes that will be noticeable, so I think we should wait a few days and update the screenshots for consistency

J2-D2-3PO commented:
@andrewtruong @bcsherma thanks for the review. RE your feedback, I addressed in [4b718e8](https://github.com/wandb/weave/pull/3579/commits/4b718e84e3002a85d56060e56a8e43acb51f1c43):

1. Updated "dataset" and "Dataset" back to `Dataset` since it's code / proper noun
2. @bcsherma fixed that kind of funky sentence you noted 
3. @bcsherma I understand the point about the screenshots, but I'd rather just fast follow with updates and get these docs out to the user base now so they have something to work with.

Once the CI storm clears, l'll ship this
